src/content/1.7/development/architecture/migration-guide/forms/CRUD-form.md

```diff
@@ -0,0 +1,245 @@
+```
+
+{{% notice note %}}
+ Make sure you pass your form type as a string.
+{{% /notice %}}
+
+Finally, use it in your controller:
+
+```php
+    public function editAction($contactId)
+    {
+        $contactFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.contact_form_builder');
+        $contactForm = $contactFormBuilder->getFormFor($contactId);
+
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/Contact/Contacts/edit.html.twig', [
+            'contactForm' => $contactForm->createView(),
+        ]);
+    }
+```
+
+and the result with some common twig structure applied for identifiable object form pages looks like this:
+
+{{< figure src="../img/contact-form-rendered.png" title="Result of form builder" >}}
+
+## Form data handler
+
+Form data handler is responsible for your form data submitting and it is used by these methods:
+
+* **create($data)** - used for creating new identifiable object.
+
+* **update($id, $data)** - used for updating identifiable object.
+
+### Creating form data handler
+
+You don’t have to create the Form data handler by yourself but instead rely on `\PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\FormDataHandlerInterface`
+
+```php
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+final class ContactFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        // handle creation logic here
+        return 'my required data to use after identifiable object is created';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($id, array $data)
+    {
+        /** @var ContactId $contactId */
+        $contactId = $this->updateContact($id, $data);
+
+        return $contactId->getValue();
+    }
+
+    public function updateContact($id, $data)
+    {
+        // Identifiable object update logic goes here.
+    }
+}
+```
+
+After initializing `create` method there you must handle identifiable objects creation logic. You can return any data you like or no data at all - it depends on the use case where the data will be used next.
+After initializing `update` method you must update your identifiable object and it must return an `int` type id.
+Don't forget to register it as a service because it is a part of [Form Handler](#form-handler).
+
+```yaml
+#src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml
+
+  prestashop.core.form.identifiable_object.data_handler.contact_form_data_handler:
+    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\ContactFormDataHandler'
+```
+
+{{% notice note %}}
+On real use case scenario the data is used to create or modify certain table in your database. Have a look at our [recommended approach]({{< relref "SQRS-usage-in-forms.md" >}}) of how to do that withing the _Form data handler_.
```

Comment:
in context of PrestaShop, real (or most common) use case is either use `ObjectModel` or `CQRS` so i dont think you have to be that specific about database only.